_rules/SC4-1-2-role-attribute-has-valid-value.md

```diff
@@ -0,0 +1,123 @@
+---
+name: Role attribute has valid value
+test_type: atomic
+
+description: |
+   This rule checks that each role attribute has at least one valid value
+   
+success_criterion:
+- 4.1.2 # Name, Role, Value
+
+test_aspects:
+- DOM Tree
+- CSS Styling
+
+authors:
+- Jey Nandakumar
+---
+
+## Test procedure
+
+### Applicability
+
+Any [non-empty](#non-empty) `role` attribute that is specified on an HTML or SVG element that is [exposed to assistive technologies](#exposed-to-assistive-technologies).
+
+### Expectation
+
+Each test target has at least one non-abstract valid value as specified in the [WAI-ARIA 1.1 Categorization of Roles](https://www.w3.org/TR/wai-aria-1.1/#roles_categorization).
```

Comment:
This should include roles from aria-dpub and aria-graphics. I suggest we create a separate definition where we keep a list of ARIA core + modules that should be supported.

I'd also suggest we say "valid role" instead of "valid value".

Reply:
See also https://github.com/auto-wcag/auto-wcag/pull/273#issuecomment-428152536.